spec/workers/feeds/import_articles_worker_spec.rb

```diff
@@ -2,45 +2,42 @@
 
 RSpec.describe Feeds::ImportArticlesWorker, type: :worker, sidekiq: :inline do
   let(:worker) { subject }
+  let(:feed_url) { "https://medium.com/feed/@vaidehijoshi" }
 
   include_examples "#enqueues_on_correct_queue", "medium_priority"
 
   describe "#perform" do
-    it "calls the Feeds::Import defaulting to 4 hours ago" do
+    it "processes jobs for users with feeds", vcr: "feeds_import" do
       allow(Feeds::Import).to receive(:call)
       alice = create(:user)
       bob = create(:user)
+      bob.setting.update(feed_url: feed_url)
+
+      # bob has a feed, and alice doesn't, so we only enqueued for bob
 
       Timecop.freeze(Time.current) do
         worker.perform
 
-        # Each user is run separately. Note that this will not be run
-        # sequentially like this in production. This only works like this due
-        # to the `sidekiq: :inline` tag on the `RSpec.describe` block
-        expect(Feeds::Import)
-          .to have_received(:call)
-          .with(users_scope: User.where(id: [alice.id]), earlier_than: 4.hours.ago.iso8601)
         expect(Feeds::Import)
           .to have_received(:call)
           .with(users_scope: User.where(id: [bob.id]), earlier_than: 4.hours.ago.iso8601)
+        expect(Feeds::Import)
+          .not_to have_received(:call)
+          .with(users_scope: User.where(id: [alice.id]), earlier_than: 4.hours.ago.iso8601)
       end
     end
 
-    it "calls the Feeds::Import with the given time" do
-      allow(Feeds::Import).to receive(:call)
-      alice = create(:user)
-      bob = create(:user)
+    it "enqueues job for user with the given time", vcr: "feeds_import", sidekiq: :fake do
+      user = create(:user)
+      user.setting.update(feed_url: feed_url)
 
       Timecop.freeze(Time.current) do
-        worker.perform([], 1.minute.ago)
+        earlier_than = 1.minute.ago
 
-        # Each user is run separately
-        expect(Feeds::Import)
-          .to have_received(:call)
-          .with(users_scope: User.where(id: [alice.id]), earlier_than: 1.minute.ago.iso8601)
-        expect(Feeds::Import)
-          .to have_received(:call)
-          .with(users_scope: User.where(id: [bob.id]), earlier_than: 1.minute.ago.iso8601)
+        sidekiq_assert_enqueued_with(job: Feeds::ImportArticlesWorker::ForUser,
+                                     args: [user.id, earlier_than.iso8601]) do
+          worker.perform([], earlier_than)
+        end
```

Comment:
Why fake out the Sidekiq work?

Reply:
good question (no good answer) - _originally_ I didn't see that we were putting a spy on call and wanted to avoid processing the feed (I see why the spec is written with sidekiq inline, particularly because it used to call directly, but other things being equal I usually test "it enqueues a job" as the boundary between this class and the next). My first intention was to only use fake sidekiq for this job, and add a quick spec on ForUser (which is simple, and doesn't do any filtering) saying it calls Feeds::Import with provided args.

This faked sidekiq is now an experiment... totally low stakes to revert, it was part of an exploration of why these tests weren't failing (it was easier for me to see what was happening when the tests weren't running any enqueued jobs, since I had problems in deciding which should be enqueued).